tools/run_tests/run_tests_matrix.py

```diff
@@ -209,100 +209,108 @@ def _allowed_labels():
+    extra_args.append('--force_default_poller')
 
-extra_args = []
-if args.build_only:
-  extra_args.append('--build_only')
-if args.force_default_poller:
-  extra_args.append('--force_default_poller')
+  all_jobs = _create_test_jobs(extra_args=extra_args) + _create_portability_test_jobs(extra_args=extra_args)
 
-all_jobs = _create_test_jobs(extra_args=extra_args) + _create_portability_test_jobs(extra_args=extra_args)
+  jobs = []
+  for job in all_jobs:
+    if not args.filter or all(filter in job.labels for filter in args.filter):
+      jobs.append(job)
 
-jobs = []
-for job in all_jobs:
-  if not args.filter or all(filter in job.labels for filter in args.filter):
-    jobs.append(job)
+  if not jobs:
+    jobset.message('FAILED', 'No test suites match given criteria.',
+                   do_newline=True)
+    sys.exit(1)
 
-if not jobs:
-  jobset.message('FAILED', 'No test suites match given criteria.',
-                 do_newline=True)
-  sys.exit(1)
-  
-print('IMPORTANT: The changes you are testing need to be locally committed')
-print('because only the committed changes in the current branch will be')
-print('copied to the docker environment or into subworkspaces.')
+  print('IMPORTANT: The changes you are testing need to be locally committed')
+  print('because only the committed changes in the current branch will be')
+  print('copied to the docker environment or into subworkspaces.')
+  print
 
-print 
-print 'Will run these tests:'
-for job in jobs:
-  if args.dry_run:
-    print '  %s: "%s"' % (job.shortname, ' '.join(job.cmdline))
-  else:
-    print '  %s' % job.shortname
-print
+  skipped_jobs = []
+  if args.filter_pr_tests:
+    print 'Looking for irrelevant tests to skip...'
+    relevant_jobs = filter_tests(jobs, args.base_branch)
+    print
+    if len(relevant_jobs) == len(jobs):
+      print 'No tests will be skipped.'
+    else:
+      print 'These tests will be skipped:'
+      skipped_jobs = set(jobs) - set(relevant_jobs)
+      for job in list(skipped_jobs):
+        print '  %s' % job.shortname
+    jobs = relevant_jobs
+    print
 
-if args.filter_pr_tests:
-  print 'IMPORTANT: Test filtering is not active; this is only for testing.'
-  relevant_jobs = filter_tests(jobs, args.base_branch)
-  # todo(mattkwong): add skipped tests to report.xml
-  print
-  if len(relevant_jobs) == len(jobs):
-    print '(TESTING) No tests will be skipped.'
-  else:
-    print '(TESTING) These tests will be skipped:'
-    for job in list(set(jobs) - set(relevant_jobs)):
+  print 'Will run these tests:'
+  for job in jobs:
+    if args.dry_run:
+      print '  %s: "%s"' % (job.shortname, ' '.join(job.cmdline))
+    else:
       print '  %s' % job.shortname
   print
 
-if args.dry_run:
-  print '--dry_run was used, exiting'
-  sys.exit(1)
+  if args.dry_run:
+    print '--dry_run was used, exiting'
+    sys.exit(1)
+
+  jobset.message('START', 'Running test matrix.', do_newline=True)
+  num_failures, resultset = jobset.run(jobs,
```

Comment:
if there are no jobs to run, what's going to happen?  (likely to happen for many PRs with sanity linux test suites). 
It shouldn't fail and it need to generate report.xml because jenkins will be looking for it (and fails the job if it's not present).